### ISSUE TITLE
Cannot use a JsonQualifier on a value class with generics

### ISSUE THREAD

**Issue by mattbdean** [rharter/auto-value-moshi#92]
Given this class:

```java
@AutoValue
public abstract class Foo<T> {
    @MyJsonQualifier
    public abstract T bar();

    public static <T> JsonAdapter<Foo<T>> jsonAdapter(Moshi moshi, Type[] types) {
        return new AutoValue_Foo.MoshiJsonAdapter(moshi, types);
    }
}
```

The presence of`bar()`'s JsonQualifier triggers the generation of another method:

```java
private JsonAdapter adapter(Moshi moshi, String methodName) {
  // ...
  Method method = Foo<T>.class.getDeclaredMethod(methodName);
  // ...
}
```

This in turn produces this error:
```
/path/to/generated/source/AutoValue_Foo.java:59: error: <identifier> expected
        Method method = Foo<T>.class.getDeclaredMethod(methodName);
                               ^
```

As far as I can tell, this error originates from [here](https://github.com/rharter/auto-value-moshi/blob/e8421aa2c038787ab2172c88f38812b859b01912/auto-value-moshi/src/main/java/com/ryanharter/auto/value/moshi/AutoValueMoshiExtension.java#L350). Full source/generated code is available in [this gist](https://gist.github.com/thatJavaNerd/ccde21601e9f7f5c5787c8dd1f74f90a).

Is there something I'm missing or is this a genuine bug?

**Comment by ZacSweers** [comment]
Looks like a genuine bug, just a new case that wasn't covered before. We'll want to remove generics from the method lookup and also make sure we're passing the generic Types we take in to the adapter to the qualifier lookup. Interested in contributing a PR?

**Comment by mattbdean** [comment]
I could certainly take a look!

**Comment by ZacSweers** [comment]
Resolved by #93
